include/swift/AST/DiagnosticsSema.def

```diff
@@ -2733,6 +2733,8 @@ ERROR(differentiable_attr_unsupported_req_kind,none,
       "layout requirement are not supported by '@differentiable' attribute", ())
 ERROR(differentiable_attr_class_unsupported,none,
       "class members cannot be marked with '@differentiable'", ())
+ERROR(differentiable_attr_stored_property_variable_unsupported,none,
+      "stored properties/variables cannot be marked with '@differentiable' with a custom VJP/JVP", ())
```

Comment:
Let me know if anyone has a better idea for a message. As we need to account for global variables/constants not being allowed to have custom VJPs/JVPs.

Reply:
1. "VJP" or "JVP" does not formally exist in AD's user vocabulary, and we've been steering the programming model away from these words. In this case, since it's unavoidable to mention these terms since the user specified the `jvp:` or `vjp:` label, it's better to just refer to them verbatim, i.e. `'jvp:' or 'vjp:' cannot be specified for stored properties`. No need to mention `'@differentiable'` in this case since the source location is at the right spot.
2. Thanks for catching the `/variables` case!

Reply:
> Thanks for catching the /variables case!

Update on this: Actually it seems as though a different error is being thrown:
```c++
// Global immutable vars, for example, have no getter, and therefore trigger
// this.
if (!original) {
  diagnoseAndRemoveAttr(attr, diag::invalid_decl_attribute, attr);
  return;
}
```
Currently the error `'@differentiable' attribute cannot be applied to this declaration` is being thrown on both global `var` variables and global `let` constants. To me, it seems as though we should be able to mark global variables as differentiable (not constants though).